lib/matplotlib/image.py

```diff
@@ -1582,6 +1584,21 @@ def imsave(fname, arr, vmin=None, vmax=None, cmap=None, format=None,
             _api.check_in_list(('upper', 'lower'), origin=origin)
         if origin == "lower":
             arr = arr[::-1]
+
+        # This specifically handled list-of-list-of-list
+        # Produced an image instance using the data from arr and do scaling
+        if (isinstance(arr, list)):
+            fig = Figure()
+            ax = fig.add_axes([0, 0, 1, 1],
+                              aspect='auto',
+                              frameon=False,
+                              xticks=[],
+                              yticks=[])
+            im = AxesImage(ax, cmap=cmap)
+            im.set_data(arr)
+            im._scale_norm(None, vmin, vmax)
+            arr = im.get_array()
```

Comment:
I'm confused: Why do we explicitly scale if the data are lists and not if they are arrays? Shouldn't they be handled equally wrt to scaling whether it's `[[1, 2]], [3, 4]]` or `np.array([[1, 2]], [3, 4]])`? Also, are we sure that this plays well with the following if/else block? Overall, from https://github.com/matplotlib/matplotlib/issues/29183#issuecomment-2504903742 I would have expected nut much more than `arr = np.asarray(arr, dtype=np.uint8)` in an appropriate place.

Additionally, note *if* separate scaling is needed, it should look more like the scaling in the else block below and not create a whole figure.

Reply:
I think we can get away with only doing `arr = np.asarray(arr)` here and dropping the `isinstance` check in the next line.